metis-enrichment/metis-enrichment-client/src/main/java/eu/europeana/enrichment/rest/client/enrichment/MetisRecordParser.java

```diff
@@ -73,26 +71,27 @@
 
     // Get all direct references (also look in Europeana proxy as it may have been dereferenced - we
     // use this below to follow sameAs links).
-    final List<AboutType> proxies = Optional.ofNullable(rdf.getProxyList()).stream()
+    final List<ProxyType> proxies = Optional.ofNullable(rdf.getProxyList()).stream()
         .flatMap(Collection::stream).filter(Objects::nonNull).collect(Collectors.toList());
-    final Map<String, Set<FieldType>> directReferences = new HashMap<>();
-    for (FieldType field : ProxyFieldType.values()) {
+    final Map<String, Set<ProxyFieldType>> directReferences = new HashMap<>();
+    for (ProxyFieldType field : ProxyFieldType.values()) {
       final Set<String> directLinks = proxies.stream().map(field::extractFieldLinksForEnrichment)
           .flatMap(Set::stream).collect(Collectors.toSet());
       for (String directLink : directLinks) {
-        directReferences.computeIfAbsent(directLink, key -> new HashSet<>())
-            .add(field);
+        directReferences.computeIfAbsent(directLink, key -> new HashSet<>()).add(field);
       }
     }
 
     // Get all sameAs links from the directly referenced contextual entities.
-    final Map<String, Set<FieldType>> indirectReferences = new HashMap<>();
+    final Map<String, Set<ProxyFieldType>> indirectReferences = new HashMap<>();
     final Consumer<AboutType> contextualTypeProcessor = contextualClass -> {
-      final Set<FieldType> linkTypes = directReferences.get(contextualClass.getAbout());
-      if (linkTypes != null) {
+      final Set<ProxyFieldType> linkTypes = Optional
+          .ofNullable(directReferences.get(contextualClass.getAbout()))
+          .orElseGet(Collections::emptySet).stream().map(ProxyFieldType.class::cast)
```

Comment:
Casting should not be necessary here... it's already the right type.

Reply:
Missed that one